tensorflow-core/tensorflow-core-api/src/gen/java/org/tensorflow/internal/c_api/global/tensorflow.java

```diff
@@ -11,6 +11,264 @@
+
+public static native @Cast("const char*") BytePointer TF_TString_GetDataPointer(@Const TF_TString str);
+
+public static native @Cast("char*") BytePointer TF_TString_ResizeUninitialized(TF_TString str,
+                                                   @Cast("size_t") long new_size);
+
+public static native @Cast("char*") BytePointer TF_TString_GetMutableDataPointer(TF_TString str);
+
+public static native void TF_TString_Reserve(TF_TString str, @Cast("size_t") long new_cap);
+
+public static native @Cast("char*") BytePointer TF_TString_Resize(TF_TString str, @Cast("size_t") long new_size,
+                                      @Cast("char") byte c);
+
+public static native void TF_TString_AssignView(TF_TString dst, @Cast("const char*") BytePointer src,
+                                         @Cast("size_t") long size);
+public static native void TF_TString_AssignView(TF_TString dst, String src,
+                                         @Cast("size_t") long size);
+
+public static native void TF_TString_AppendN(TF_TString dst, @Cast("const char*") BytePointer src,
+                                      @Cast("size_t") long src_size);
+public static native void TF_TString_AppendN(TF_TString dst, String src,
+                                      @Cast("size_t") long src_size);
+
+public static native void TF_TString_Append(TF_TString dst, @Const TF_TString src);
+
+public static native void TF_TString_Copy(TF_TString dst, @Cast("const char*") BytePointer src,
+                                   @Cast("size_t") long size);
+public static native void TF_TString_Copy(TF_TString dst, String src,
+                                   @Cast("size_t") long size);
+
+public static native void TF_TString_Assign(TF_TString dst, @Const TF_TString src);
+
+public static native void TF_TString_Move(TF_TString dst, TF_TString src);
+
+// #endif  // TENSORFLOW_CORE_PLATFORM_CTSTRING_INTERNAL_H_
+
+
+// Parsed from tensorflow/core/platform/ctstring.h
+
+/* Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+// #ifndef TENSORFLOW_CORE_PLATFORM_CTSTRING_H_
+// #define TENSORFLOW_CORE_PLATFORM_CTSTRING_H_
+
+// #include <stdint.h>
+// #include <stdlib.h>
+
+// #include "tensorflow/core/platform/ctstring_internal.h"
+
+// Initialize a new tstring.  This must be called before using any function
```

Comment:
Any clue why the documentation of these new string endpoints get all messed up by being aggregated together?

Reply:
The TF_TString type isn't defined in ctstring.h, so we need to include ctstring_internal.h as well, which contains the definitions of those functions too. JavaCPP simply skips the redundant declarations that follow. We can hack something up, but this is something they should fix upstream, and eventually will I assume, so probably not something worth spending time on here.